pkg/csi/service/common/vsphereutil.go

```diff
@@ -171,6 +170,43 @@ func CreateBlockVolumeUtil(ctx context.Context, clusterFlavor cnstypes.CnsCluste
 				spec.ScParams.DatastoreURL)
 		}
 	}
+	if spec.StoragePolicyID != "" {
+		// Check storage policy compatibility.
+		var sharedDSMoRef []vim25types.ManagedObjectReference
+		for _, ds := range datastores {
+			sharedDSMoRef = append(sharedDSMoRef, ds.Reference())
```

Comment:
In the case of topology deployment, does this variable - `datastores` holds a list of shared datastore accessible to nodes belonging to the topology requirements supplied in the creates volume call?

Since the title of the PR mentions "all VMs", we need to make sure we do not break the topology feature.

Reply:
Couple of comments:
1. In a topology aware environment, this exact code is already run for preferential datastores. Can we find a way to skip it? Reference: https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/csi/service/common/commonco/k8sorchestrator/topology.go#L958-L994
2. This function is called from WCP as well. Is it required in WCP and did you test if it works fine there?

Reply:
Tested with topology setup and for WCP as well